File: app/src/main/java/com/jawnnypoo/openmeh/receiver/BootReceiver.java
package com.jawnnypoo.openmeh.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.jawnnypoo.openmeh.util.MehPreferencesManager;
import com.jawnnypoo.openmeh.util.MehReminderManager;

/**
 * Boot receiver so that we can restore alarms when the phone boots.
 *
 * Keep in mind that the BootReceiver does not work for application
 * installs on external storage
 */
public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        restoreAlarms(context);
    }

    private void restoreAlarms(Context context) {
        if (MehPreferencesManager.getNotificationsPreference(context)) {
            MehReminderManager.restoreReminderPreference(context);
        }
    }
}

File: app/src/main/java/com/jawnnypoo/openmeh/receiver/BootReceiver.kt
package com.jawnnypoo.openmeh.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.jawnnypoo.openmeh.util.MehReminderManager
import com.jawnnypoo.openmeh.util.Prefs

/**
 * Boot receiver so that we can restore alarms when the phone boots.

 * Keep in mind that the BootReceiver does not work for application
 * installs on external storage
 */
class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        restoreAlarms(context)
    }

    private fun restoreAlarms(context: Context) {
        if (Prefs.getNotificationsPreference(context)) {
            MehReminderManager.restoreReminderPreference(context)
        }
    }
}